### ISSUE TITLE
Improve checks on various CCE rules

### ISSUE THREAD

**Issue by tdrimmelen** [microsoft/OMS-Agent-for-Linux#1195]
Hi,

CCE checks for SSH config can be improved as they no give false positives. Seen that with rules:
- CCE-4730-3
- CCE-4030-3
- CCE-3660-8
- CCE-4030-3

All rules give a 'failed' when the configuration line in sshd_config is missing although the default behaviour of the SSH daemon when the configuration option is absent, is the intended one.

Propose to improve the logic that test passes either:
- The configuration item is presented with the required value
- The configuration is absent

For example (CCE-4030-3):
- If the file contains 'RhostsAuthentication = no', the check passes
- If the file does not contains an extra with 'RhostsAuthentication', the check passes
- If the file contains 'RhostsAuthentication = yes' the check fails

In other words:
The checks fails when 'RhostsAuthentication = yes' is found.

Regards,

Theo

**Comment by hestolz** [comment]
Thank you for your submission. As [previously announced](https://azure.microsoft.com/en-us/updates/were-retiring-the-log-analytics-agent-in-azure-monitor-on-31-august-2024/), the Log Analytics agent has been **deprecatated** and has no support as of **August 31, 2024.** If you use the Log Analytics agent to ingest data to Azure Monitor, [migrate now to the new Azure Monitor agent](https://docs.microsoft.com/en-us/azure/azure-monitor/agents/azure-monitor-agent-migration). As part of repo archival, open issues and pull requests will be closed.
